core/src/OpenMP/Kokkos_OpenMP_Instance.cpp

```diff
@@ -517,7 +515,3 @@ constexpr DeviceType DeviceTypeTraits<OpenMP>::id;
 #endif
 
 }  // namespace Kokkos
-
-#else
-void KOKKOS_CORE_SRC_OPENMP_EXEC_PREVENT_LINK_ERROR() {}
-#endif  // KOKKOS_ENABLE_OPENMP
```

Comment:
This is the only change except for the renaming. If `KOKKOS_ENABLE_OPENMP` is false, the `OpenMP` directory is not included, so this is not necessary.